src/common_format.rs

```diff
@@ -117,72 +118,83 @@ impl Calendar {
     }
 }
 
-fn manage_calendar_dates(
+fn manage_calendar_dates<'a, H>(
     calendars: &mut CollectionWithId<objects::Calendar>,
-    path: &path::Path,
-) -> Result<()> {
+    file_handler: &'a mut H,
+) -> Result<()>
+where
+    &'a mut H: FileHandler,
+{
     let file = "calendar_dates.txt";
-    let path = path.join(file);
-    if !path.exists() {
-        info!("Skipping {}", file);
-    } else {
-        info!("Reading {}", file);
-
-        let mut rdr = csv::Reader::from_path(&path).with_context(ctx_from_path!(path))?;
-        for calendar_date in rdr.deserialize() {
-            let calendar_date: CalendarDate = calendar_date.with_context(ctx_from_path!(path))?;
-
-            let is_inserted = calendars
-                .get_mut(&calendar_date.service_id)
-                .map(|mut calendar| match calendar_date.exception_type {
-                    ExceptionType::Add => {
-                        calendar.dates.insert(calendar_date.date);
-                    }
-                    ExceptionType::Remove => {
-                        calendar.dates.remove(&calendar_date.date);
+
+    let reader = file_handler.get_file(file);
+    match reader {
+        Err(_) => info!("Skipping {}", file),
+        Ok(reader) => {
+            info!("Reading {}", file);
+
+            let mut rdr = csv::Reader::from_reader(reader);
+            for calendar_date in rdr.deserialize() {
+                let calendar_date: CalendarDate =
+                    calendar_date.with_context(ctx_from_filename!(file))?;
```

Comment:
indeed, it would be a better to have a full path with the name of the content file causing problem. Deal ?

Reply:
gnagnagna mutlitple mutable borrow, gnagnagna :gun:

Reply:
yes that's what I did finally, because spliting the method is not possbile

I thus Returns `Result<(Option<Reader>, PathBuf)>` is that ok for you ? (did not see the comment on `get_file_exists` before :roll_eyes: , but the use is not that bad)